### PR TITLE
Pin Transformers to 4.31.0

### PR DESCRIPTION
**Pull request by arnavgarg1**
It seems like the latest `transformers` package has a bug. This can be fixed by either using `transformers` master, or downgrading to `4.31.0` which is stable and works correctly. 

This is the error we run into with 4.32.1:
```
site-packages/ludwig/api.py:1510), in LudwigModel.preprocess(self, dataset, training_set, validation_set, test_set, training_set_metadata, data_format, skip_save_processed_input, random_seed, **kwargs)
   1508     return PreprocessedDataset(proc_training_set, proc_validation_set, proc_test_set, training_set_metadata)
   1509 except Exception as e:
-> 1510     raise RuntimeError(f"Caught exception during model preprocessing: {str(e)}") from e
   1511 finally:
   1512     for callback in self.callbacks:
RuntimeError: Caught exception during model preprocessing: local variable 'tokens' referenced before assignment
```

Closes: #3568

### CODE REVIEW

**Review by tgaddair — APPROVED** [review]
Can you file an issue with HF transformers? It might be an API issue on our end, but either way we probably need them to weigh in.

Also, please add a TODO to the requirements file to revert this once the issue is resolved.

File: requirements.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ torchaudio
 torchtext
 torchvision
 pydantic<2.0
-transformers>=4.31.0
+transformers<4.32.0
```
Comment (tgaddair) [inline requirements.txt:14]: Should be:

```
transformers>=4.31.0;<4.32.0
```

**Review by tgaddair — CHANGES_REQUESTED** [review]
Please fix lower bound.

**Comment by arnavgarg1** [comment]
@tgaddair seems like the issue was flagged and fixed on `transformers` master: https://github.com/huggingface/transformers/issues/25805

**Comment by arnavgarg1** [comment]
Associated Ludwig issue: https://github.com/ludwig-ai/ludwig/issues/3571

**Comment by arnavgarg1** [comment]
Associated Ludwig issue: https://github.com/ludwig-ai/ludwig/issues/3571

**Review by justinxzhao — APPROVED** [review]

**Review by tgaddair — APPROVED** [review]

**Pull request merged**